components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/StringResources.kt

```diff
@@ -0,0 +1,153 @@
+            val strings = nodes.getElementsWithName("string").associate { element ->
+                element.getAttribute("name") to StringItem.Value(element.textContent.orEmpty())
+            }
+            val arrays = nodes.getElementsWithName("string-array").associate { arrayElement ->
+                val items = arrayElement.childNodes.getElementsWithName("item").map { element ->
+                    element.textContent.orEmpty()
+                }
+                arrayElement.getAttribute("name") to StringItem.Array(items)
+            }
+            strings + arrays
+        }
+    }
+
+/**
+ * Retrieves a string resource using the provided ID.
+ *
+ * @param id The ID of the string resource to retrieve.
+ * @return The retrieved string resource.
+ *
+ * @throws IllegalArgumentException If the provided ID is not found in the resource file.
+ */
+@ExperimentalResourceApi
+@Composable
+fun getString(id: ResourceId): String {
+    val resourceReader = LocalResourceReader.current
+    val str by rememberState(id, "") { loadString(id, resourceReader) }
+    return str
+}
+
+/**
+ * Loads a string resource using the provided ID.
+ *
+ * @param id The ID of the string resource to load.
+ * @return The loaded string resource.
+ *
+ * @throws IllegalArgumentException If the provided ID is not found in the resource file.
+ */
+@ExperimentalResourceApi
+suspend fun loadString(id: ResourceId): String = loadString(id, DefaultResourceReader)
+
+private suspend fun loadString(id: ResourceId, resourceReader: ResourceReader): String {
+    val nameToValue = getParsedStrings(getPathById(STRINGS_XML), resourceReader)
+    val item = nameToValue[id.stringKey] as? StringItem.Value
+        ?: error("String ID=`${id.stringKey}` is not found!")
+    return item.text
+}
+
+/**
+ * Retrieves a formatted string resource using the provided ID and arguments.
+ *
+ * @param id The ID of the string resource to retrieve.
+ * @param formatArgs The arguments to be inserted into the formatted string.
+ * @return The formatted string resource.
+ *
+ * @throws IllegalArgumentException If the provided ID is not found in the resource file.
+ */
+@ExperimentalResourceApi
+@Composable
+fun getString(id: ResourceId, vararg formatArgs: Any): String {
+    val resourceReader = LocalResourceReader.current
+    val args = formatArgs.map { it.toString() }
+    val str by rememberState(id, "") { loadString(id, args, resourceReader) }
+    return str
+}
+
+/**
+ * Loads a formatted string resource using the provided ID and arguments.
+ *
+ * @param id The ID of the string resource to load.
+ * @param formatArgs The arguments to be inserted into the formatted string.
+ * @return The formatted string resource.
+ *
+ * @throws IllegalArgumentException If the provided ID is not found in the resource file.
+ */
+@ExperimentalResourceApi
+suspend fun loadString(id: ResourceId, vararg formatArgs: Any): String = loadString(
```

Comment:
Why do we need it in public?

Reply:
To be able to load strings from non-UI components. (e.g. ViewModels) It is a common case.